src/plugins/ui_stateful/edition.ts

```diff
@@ -164,41 +221,40 @@ export class EditionPlugin extends UIPlugin {
         break;
       case "START_CHANGE_HIGHLIGHT":
         this.dispatch("STOP_COMPOSER_RANGE_SELECTION");
+        const state = this.state as EditingState;
         const range = this.getters.getRangeFromRangeData(cmd.range);
-        const previousRefToken = this.currentTokens
+        const previousRefToken = this.state.currentTokens
           .filter((token) => token.type === "REFERENCE")
           .find((token) => {
             let value = token.value;
             const [xc, sheet] = value.split("!").reverse();
-            const sheetName = sheet || this.getters.getSheetName(this.sheetId);
+            const sheetName = sheet || this.getters.getSheetName(state.sheetId);
             const activeSheetId = this.getters.getActiveSheetId();
             if (this.getters.getSheetName(activeSheetId) !== sheetName) {
               return false;
             }
             const refRange = this.getters.getRangeFromSheetXC(activeSheetId, xc);
             return isEqual(this.getters.expandZone(activeSheetId, refRange.zone), range.zone);
           });
-        this.previousRef = previousRefToken!.value;
-        this.previousRange = this.getters.getRangeFromSheetXC(
+        this.state.previousRef = previousRefToken!.value;
+        this.state.previousRange = this.getters.getRangeFromSheetXC(
           this.getters.getActiveSheetId(),
-          this.previousRef
+          this.state.previousRef
         );
-        this.selectionInitialStart = previousRefToken!.start;
-        break;
-      case "CHANGE_HIGHLIGHT":
-        const cmdRange = this.getters.getRangeFromRangeData(cmd.range);
-        const newRef = this.getRangeReference(cmdRange, this.previousRange!.parts);
-        this.selectionStart = this.selectionInitialStart;
-        this.selectionEnd = this.selectionInitialStart + this.previousRef.length;
-        this.replaceSelection(newRef);
-        this.previousRef = newRef;
-        this.selectionStart = this.currentContent.length;
-        this.selectionEnd = this.currentContent.length;
+        this.state.selectionInitialStart = previousRefToken!.start;
```

Comment:
the 5 previous lines now need to be called by this handle ('START_CHANGE_HIGHLIGHTS")  but also when handling 2 selection events ..; and also when activating a new sheet ....  It turns out that we do not really need to store this reference and range because their value and position can actually be infered during the handling of the selection event (since we have previous zone and the new zone in the payload) but this mean doing all the filter and find token everytime we handle a selection event.

Since the code needs to filter and then find in array with a rather 'heavy' condition, I feel like it will not scale that much if we were to call it everytime we change the selection ...  

@LucasLefevre do you have an opinion on the matter ?

Reply:
not sure to understand what you mean.
I think it's a hidden message from my brain telling me to stop working for now
I'll try again tomorrow

Reply:
We can report this discussion for another task/pr - I realized that we are supposed (?) to support unbounded zone in the resize stuff but we are not really doing it that atm because the Highlight composer don't. 

since selection processor does not handle those zones (yet) we can leave it for another improvement